docs/fundamentals/runtime-libraries/system-reflection-emit-assemblybuilder.md

```diff
@@ -119,6 +132,124 @@ using var fileStream = new FileStream("MyAssembly.exe", FileMode.Create, FileAcc
 peBlob.WriteContentTo(fileStream); 
 ```
 
+### Emitting symbols and generating PDB
+
+The symbols metadata populated into the `pdbBuilder` out parameter when `GenerateMetadata(out BlobBuilder ilStream, out BlobBuilder mappedFieldData, out MetadataBuilder pdbBuilder)` method called on `PersistedAssemblyBuilder` instance. The steps for creating an assembly with portable PDB:
+
+1. Create `ISymbolDocumentWriter` instances with `ModuleBuilder.DefineDocument(...)` method, while emitting method's IL also emit the corresponding symbol info.
+2. Create `PortablePdbBuilder` instance using the `pdbBuilder` instance produced with `GenerateMetadata(...)` method.
+3. Serialize the `PortablePdbBuilder` into a `Blob`, write the `Blob` into a PDB file stream (only in case generating standalone PDB).
+4. Create `DebugDirectoryBuilder` instance and add a `CodeViewEntry` (standalone PDB) or `EmbeddedPortablePdbEntry`.
+5. Set the optional `debugDirectoryBuilder` argument when creating `PEBuilder` instance.
+
+Following example shows how to emit symbol info and generate PDB:
+
+```csharp
+static void GenerateAssemblyWithPdb()
```

Comment:
It would be better to put this into a separate code file with an associated project file so we can check compilation as part of this PR.

Reply:
Did not know this repo also can compile, anyway these are mostly partial codes, it might compile as is but not runnable, I plan to add the compile able and runnable examples in dotnet-api-docs repo for the specific APIs. If you still prefer to have at least compile able code I can make that happen.

Also, I was wondering if we should have a separate doc for `PersistedAssemblyBuilder` as it has got lot more content than other `AssemblyBuilder` implementation and even have a different name, what you think?

Reply:
> Also, I was wondering if we should have a separate page for PersistedAssemblyBuilder as it has got lot more content than other AssemblyBuilder implementation and even have a different name, what you think?

Yes, good point. Let's create a new file for it (using the same naming pattern). You or I should also a link to it from the Remarks section of https://learn.microsoft.com/en-us/dotnet/api/system.reflection.emit.persistedassemblybuilder?view=net-9.0#applies-to.

Reply:
If we can at least compile the examples here, that's great.

Reply:
> If we can at least compile the examples here, that's great.

I moved ones that can build with .NET 9 preview 3, remaining's need to wait until preview 4 released

>  You or I should also a link to it from the Remarks section of https://learn.microsoft.com/en-us/dotnet/api/system.reflection.emit.persistedassemblybuilder?view=net-9.0#applies-to.

I can do that later while adding the examples to `dotnet-api-docs`